express-post-json/index.js

```diff
@@ -0,0 +1,32 @@
+const express = require('express');
+const app = express();
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('Express server listening on port 3000');
+});
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const array = [];
+  for (const key in grades) {
+    array.push(grades[key]);
+  }
+  res.json(array);
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  grades[nextId] = {
+    id: nextId,
+    name: req.body.name,
+    course: req.body.course,
+    score: req.body.score
+  };
+  res.status(201);
+  res.json(grades[nextId]);
+  nextId++;
+});
```

Comment:
In general, it's best to send the response after everything else is finished, so the client knows that it worked as expected.